webpack/assets/javascripts/react_app/components/statistics/ChartBox.js

```diff
@@ -29,64 +31,65 @@ class ChartBox extends React.Component {
   }
 
   render() {
-    const { chart } = this.props;
-
-    const modalConfig = getLargePieChartConfig({
-      data: this.props.chart.data,
-      id: `${chart.id}Modal`,
-    });
-
+    const { chart, type } = this.props;
+    const components = {
+      donut: DonutChart,
+    };
+    const Chart = components[type];
     const tooltip = {
       onClick: this.onClick,
       title: this.props.tip,
       'data-toggle': 'tooltip',
       'data-placement': 'top',
+      className: 'pointer',
     };
-    const onclickChartClicked =
-      chart.search && chart.search.match(/=$/) ? null : navigateToSearch.bind(null, chart.search);
-
-    const _chart = (
-      <PieChart
-        key={`${this.props.chart.id}-chart`}
-        data={this.props.chart.data}
-        onclick={onclickChartClicked}
-      />
-    );
-
+    const handleChartClick =
+      chart.search && chart.search.match(/=$/)
+        ? null
+        : navigateToSearch.bind(null, chart.search);
+    const chartProps = {
+      data: chart.data ? Immutable.asMutable(chart.data) : null,
+      key: `${this.props.chart.id}-chart`,
+      onclick: handleChartClick,
+    };
+    const panelChart = <Chart {...chartProps} />;
     const error = (
       <MessageBox
         msg={this.props.errorText}
         key={`${this.props.chart.id}-error`}
         icontype="error-circle-o"
       />
     );
+    const dataFiltered = chart.data && chart.data.filter(arr => arr[1] !== 0);
+    const hasChartData = dataFiltered && dataFiltered.length > 0;
+    const boxHeader = <h3 {...hasChartData && tooltip}>{this.props.title}</h3>;
```

Comment:
tbh, i don't like this `{...hasChartData && tooltip}` it doesn't clear that you apply all the tooltip props.
what about:
```js
const headerProps = hasChartData ? { ... } : {};
<h3 {...headerProps}>
```